Analyzer.py

```diff
@@ -264,21 +272,30 @@ def main():
             ((gemPropHit["prophit_cosine"] <= GE11_cos_phi_boundaries[1]) & (gemPropHit["prophit_cosine"] > GE11_cos_phi_boundaries[0]))* 1 +
             (gemPropHit["prophit_cosine"] > GE11_cos_phi_boundaries[1]) * 2
         )
+
+
+        # if I know the strip number i should get the "phiP" easily as the integer devision by 128 for GE11 - how is GE21 mapped?
+ 
         gemPropHit["mu_propagated_phiP"] = ak.values_astype(gemPropHit["mu_propagated_phiP"], np.short)
-        gemPropHit["mu_propagated_VFAT"] = iEtaiPhi_2_VFAT(gemPropHit.mu_propagated_etaP, gemPropHit.mu_propagated_phiP)
+        #gemPropHit["mu_propagated_VFAT"] = iEtaiPhi_2_VFAT(gemPropHit.mu_propagated_etaP, gemPropHit.mu_propagated_phiP)
+        
+        gemPropHit["mu_propagated_VFAT"] = iEtaStrip_2_VFAT(gemPropHit.mu_propagated_etaP, gemPropHit.mu_propagated_strip, gemPropHit.mu_propagated_station)
+        gemPropHit["mu_propagated_chamberType"]= iEta_2_chamberType(gemPropHit.mu_propagated_etaP,gemPropHit.mu_propagated_layer, gemPropHit.mu_propagated_station)
         gemRecHit["gemRecHit_VFAT"] = recHit2VFAT(
             gemRecHit.gemRecHit_etaPartition,
             gemRecHit.gemRecHit_firstClusterStrip,
             gemRecHit.gemRecHit_cluster_size,
+            gemRecHit.gemRecHit_station
         )
+        gemRecHit["gemRecHit_chamberType"] = iEta_2_chamberType(gemRecHit.gemRecHit_etaPartition, gemRecHit.gemRecHit_layer, gemRecHit.gemRecHit_station)
         # GE21 approach to VFAT extrapolation TO BE TESTED
         # gemPropHit["propagated_offset_phi"] =  np.arcsin(  gemPropHit.mu_propagatedLoc_x/gemPropHit.mu_propagatedGlb_r )
         # gemPropHit["propagated_VFAT"] =  (11 - (gemPropHit.propagated_offset_phi - ak.min(gemPropHit.propagated_offset_phi)) // (GE21_phi_range/6)) - ((gemPropHit.mu_propagated_etaP-1)%4)//2
         # gemRecHit["gemRecHit_VFAT"] = (11 - ((gemRecHit.gemRecHit_firstClusterStrip+gemRecHit.gemRecHit_cluster_size/2)//64)) - ((gemRecHit.gemRecHit_etaPartition - 1)%4)//2
         heap_size(the_heap, "after extrapolating on VFATs")
 
         
-        
+        #why do i need to the following? don't I know already the VFAT?
```

Comment:
To ensure that the fiducial cuts are properly applied. In particular to make sure that [this line](https://github.com/fraivone/EfficiencyAnalyzer/blob/17128f6ada61187c558a0c6f04c7484e14134807/tools/MaskFunctions.py#L56) works as expected

Reply:
Yes, I've discovered the sense of the code :D 
Anyway, probably to be revised at the same time of the fiducial cuts.